lib/mm_game.rb

```diff
@@ -0,0 +1,125 @@
+require_relative 'mm_state'
+require_relative 'mm_outputs'
+require_relative 'mm_choice'
+require_relative 'mm_input'
+require_relative 'mm_code'
+require_relative 'mm_commands'
+
+DEFAULT = {
+    state: State,
+    game_in: Input,
+    game_out: Outputs,
+    choice: Choice,
+    code: Code,
+    commands: Commands
+}
+
+class Game
+    def initialize(options = {})
+        options = DEFAULT.merge(options)
+        @status = options[:state].new
+        @in = options[:game_in]
+        @out = options[:game_out]
+        @choice = options[:choice]
+        @code = options[:code]
+        @cmds = options[:commands]
+        @input_code = nil
+        @stop_game = false
+        @game_complete = false
+        play
```

Comment:
It's weird to have the play method called in the constructor. Why not just call it after making it?